internal/controller/pdbwatcher_controller.go

```diff
@@ -37,26 +26,52 @@ type PDBWatcherReconciler struct {
 // +kubebuilder:rbac:groups=apps.mydomain.com,resources=pdbwatchers/status,verbs=get;update;patch
 // +kubebuilder:rbac:groups=apps.mydomain.com,resources=pdbwatchers/finalizers,verbs=update
 
-// Reconcile is part of the main kubernetes reconciliation loop which aims to
-// move the current state of the cluster closer to the desired state.
-// TODO(user): Modify the Reconcile function to compare the state specified by
-// the PDBWatcher object against the actual cluster state, and then
-// perform operations to make the cluster state reflect the state specified by
-// the user.
-//
-// For more details, check Reconcile and its Result here:
-// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.18.2/pkg/reconcile
 func (r *PDBWatcherReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	logger := log.FromContext(ctx)
+
+	// Fetch the PDBWatcher instance
+	pdbWatcher := &myappsv1.PDBWatcher{}
+	err := r.Get(ctx, req.NamespacedName, pdbWatcher)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+
+	// Fetch the PDB
+	pdb := &policyv1.PodDisruptionBudget{}
+	err = r.Get(ctx, types.NamespacedName{Name: pdbWatcher.Name, Namespace: pdbWatcher.Namespace}, pdb)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// Check the DisruptionsAllowed field
+	if pdb.Status.DisruptionsAllowed == 0 {
+		// Fetch the Deployment
+		deployment := &appsv1.Deployment{}
```

Comment:
TODO need to save original replicas size to know you already scaled it up 

currently if new pods don't become ready you may scale up infinitely